pkg/snapshotter/snapshotter.go

```diff
@@ -101,13 +101,37 @@ func (s *snapshot) DeleteSnapshot(ctx context.Context, snapshotID string, snapsh
 	return nil
 }
 
+func (s *snapshot) isListSnapshotsSupported(ctx context.Context) (bool, error) {
+	client := csi.NewControllerClient(s.conn)
+
+	capRsp, err := client.ControllerGetCapabilities(ctx, &csi.ControllerGetCapabilitiesRequest{})
+	if err != nil {
+		return false, err
+	}
+
+	for _, cap := range capRsp.Capabilities {
+		if cap.GetRpc().GetType() == csi.ControllerServiceCapability_RPC_LIST_SNAPSHOTS {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
 func (s *snapshot) GetSnapshotStatus(ctx context.Context, snapshotID string) (bool, time.Time, int64, error) {
 	client := csi.NewControllerClient(s.conn)
 
+	// If the driver does not support ListSnapshots, assume the snapshot ID is valid.
+	listSnapshotsSupported, err := s.isListSnapshotsSupported(ctx)
+	if err != nil {
```

Comment:
We could also just return `return true, time.Time{}, 0, nil` here, but that seems like it could cause other problems.

Also, if the GetCapabilities call fails, I figured the caller should know.

Reply:
I'm fine with this.